src/player_compchecks.c

```diff
@@ -358,7 +358,7 @@ long player_list_creature_filter_best_for_sacrifice(const struct Thing *thing, M
             return -1;
         if ((param->plyr_idx >= 0) && (thing->owner != param->plyr_idx))
             return -1;
-        if ((param->model_id > 0) && (thing->model != param->model_id))
+        if (!creature_matches_model(thing, param->model_id))
```

Comment:
I dont like that because `class_id` checked after `creature_matches_model`
what if `param->class_id` is `TCls_Object` instead of `TCls_Creature`?

Consider add some kind of `assert (thing->model == TCls_Creature)` and check where this filter actually used?

Reply:
ok, yes, the function is specifically for creatures, but I'll change this instance to thing_matches_model.